diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -183,28 +189,36 @@ controller_interface::return_type DiffDriveController::update(
   }
   else
   {
-    double left_position_mean = 0.0;
-    double right_position_mean = 0.0;
+    double left_feedback_mean = 0.0;
+    double right_feedback_mean = 0.0;
     for (size_t index = 0; index < wheels.wheels_per_side; ++index)
     {
-      const double left_position = registered_left_wheel_handles_[index].position.get().get_value();
-      const double right_position =
-        registered_right_wheel_handles_[index].position.get().get_value();
+      const double left_feedback = registered_left_wheel_handles_[index].feedback.get().get_value();
+      const double right_feedback =
+        registered_right_wheel_handles_[index].feedback.get().get_value();
 
-      if (std::isnan(left_position) || std::isnan(right_position))
+      if (std::isnan(left_feedback) || std::isnan(right_feedback))
       {
         RCLCPP_ERROR(
-          logger, "Either the left or right wheel position is invalid for index [%zu]", index);
+          logger, "Either the left or right wheel %s is invalid for index [%zu]", feedback_type(),
+          index);
         return controller_interface::return_type::ERROR;
       }
 
-      left_position_mean += left_position;
-      right_position_mean += right_position;
+      left_feedback_mean += left_feedback;
+      right_feedback_mean += right_feedback;
     }
-    left_position_mean /= wheels.wheels_per_side;
-    right_position_mean /= wheels.wheels_per_side;
+    left_feedback_mean /= wheels.wheels_per_side;
+    right_feedback_mean /= wheels.wheels_per_side;
 
-    odometry_.update(left_position_mean, right_position_mean, current_time);
+    if (odom_params_.position_feedback)
+    {
+      odometry_.update(left_feedback_mean, right_feedback_mean, current_time);
+    }
+    else
+    {
+      odometry_.updateFromVelocity(left_feedback_mean, right_feedback_mean, current_time);
```

Comment:
Why not use `period` instead of calculating it manually?

```suggestion
      odometry_.updateFromVelocity(left_feedback_mean, right_feedback_mean, period);
```

Reply:
You mean calculating 'const double dt = time.seconds() - timestamp_.seconds();' outside updateFromVelocity() and update() instead of inside? This will remove redundancy but still allow `(dt < 0.0001)` check for update() call

Reply:
we should have `period` available in the `update` method